lib/ansible/plugins/inventory/foreman.py

```diff
@@ -193,10 +152,42 @@ def to_safe(self, word):
         regex = r"[^A-Za-z0-9\_]"
         return re.sub(regex, "_", word.replace(" ", ""))
 
-    def _populate(self):
+    def _resolve_params(self, host):
+        '''
+        Fetch host params and convert to dict, then add to inventory
+        '''
+        for param in host['all_parameters']:
+            key = param['name']
+
+            # Try yaml
+            try:
+                value = yaml.load(param['value'])
+            except yaml.YAMLError:
```

Comment:
json is  a subset of yaml so it is most likely to be loaded here even if it was json to begin with, but pyyaml parser can introduce some differences, you might want to reverse the order.